variables.tf

```diff
@@ -45,3 +45,49 @@ variable "availability_zones" {
   type        = list(any)
   default     = []
 }
+
+
+
+variable "enable_flow_log" {
+  description = "Enable flow logs"
+  type        = bool
+  default     = false
+}
+
+
+
+variable "flow_log_cloudwatch_log_group_retention_in_days" {
+  description = "Retention days cloudwatch logs"
+  type        = number
+  default     = 90
+}
+
+
+
+variable "flow_log_traffic_type" {
+  description = "Traffic type (ACCEPT, REJECT, ALL)"
+  type        = string
+  default     = "ALL"
+}
+
+
+
+variable "flow_log_destination_arn" {
```

Comment:
no, the idea is, if you have a role , that you want to use, you can provide it, otherwise it will create automatically.

`  
create_flow_log_cloudwatch_iam_role             = var.enable_flow_log && var.flow_log_cloudwatch_iam_role_arn == ""
`

the same is for cloudwatch log group: if you have one , you can provide the arn , otherwise it will create new 



`create_flow_log_cloudwatch_log_group            = var.enable_flow_log && var.flow_log_destination_arn == ""`